buildpacks/jvm-function-invoker/src/error.rs

```diff
@@ -0,0 +1,123 @@
+use indoc::{formatdoc, indoc};
+use libcnb::{Error, TomlFileError};
+
+use crate::layers::bundle::BundleLayerError;
+use crate::layers::opt::OptLayerError;
+use crate::layers::runtime::RuntimeLayerError;
+use libherokubuildpack::log_error;
+
+#[derive(thiserror::Error, Debug)]
+pub enum JvmFunctionInvokerBuildpackError {
+    #[error("Opt layer error: {0}")]
+    OptLayerError(#[from] OptLayerError),
+
+    #[error("Runtime layer error: {0}")]
+    RuntimeLayerError(#[from] RuntimeLayerError),
+
+    #[error("Bundle layer error: {0}")]
+    BundleLayerError(#[from] BundleLayerError),
+
+    #[error("Could not write launch.toml: {0}")]
+    CouldNotWriteLaunchToml(TomlFileError),
+}
+
+impl From<JvmFunctionInvokerBuildpackError> for Error<JvmFunctionInvokerBuildpackError> {
+    fn from(error: JvmFunctionInvokerBuildpackError) -> Self {
+        Self::BuildpackError(error)
+    }
+}
+
+pub fn handle_buildpack_error(error: JvmFunctionInvokerBuildpackError) -> i32 {
+    match error {
+        JvmFunctionInvokerBuildpackError::OptLayerError(inner) => match inner {
+            OptLayerError::CouldNotCopyRunSh(io_error)
+            | OptLayerError::CouldNotSetExecutableBitForRunSh(io_error) => log_error(
+                "Unexpected Error",
+                formatdoc! {"
+                    An error occurred while copying files from the buildpack directory to the container.
+                    {io_error}
+                ", io_error = io_error},
```

Comment:
I didn't know you could use named arguments in formatting. I wasn't sure if it was a feature of indoc or not and it looks like it's plain old rust.

```
fn main() {
  let lol = String::from("heheh");
  println!("{lol}", lol = lol);
}
```